src/main/java/service/FruitBalanceService.java

```diff
@@ -0,0 +1,46 @@
+package service;
+
+import exceptions.BalanceException;
+import java.util.HashMap;
+import java.util.Map;
+
+public class FruitBalanceService implements BalanceService<String, Integer> {
+    private static final Map<String, Integer> fruitsBalance = new HashMap<>();
```

Comment:
please move Storage into a separate class
and create a Fruit model that you will store as a key